tsdb/tsdb.go

```diff
@@ -154,3 +155,23 @@ func (c Chain) Save() bool {
 	}
 	return true
 }
+
+//GetPositionalPointerNormalized Returns block by searching the chain for the NormalizedTime
+func (c Chain) GetPositionalPointerNormalized(n int64) *Block {
+	c.lengthElements = len(c.chain)
+	jumpSize := int(math.Floor(math.Sqrt(float64(c.lengthElements))))
+	index := jumpSize - 1
+	for n > c.chain[index].NormalizedTime {
+		index += jumpSize
```

Comment:
What if the `n` value is not present in the array. Suppose a case where the `n` value is given much larger than the possible time in `time.UnixNano()`, in that case it would lead to an infinite loop and ultimately towards a core dump which is fatal. Pls add a check in this case and report an error and also have a test case checking this whether the code throws error or not when the value is much larger than the possible `time.UnixNano()`.
Thanks!

Reply:
Doing that